src/ck-core/cklocation.C

```diff
@@ -2375,11 +2404,18 @@ CkLocMgr::CkLocMgr(CkArrayOptions opts)
 
   // Register with the map object
   mapID = opts.getMap();
-  map = (CkArrayMap*)CkLocalBranch(mapID);
-  if (map == NULL)
+  map = static_cast<CkArrayMap*>(CkLocalBranch(mapID));
+  if (map == nullptr)
     CkAbort("ERROR! Local branch of array map is NULL!");
+  // TODO: Should this be registered here, CkArray, CkLocMgr, or none?
   mapHandle = map->registerArray(opts.getEnd(), thisgroup);
 
+  cacheID = opts.getLocationCache();
+  cache = static_cast<CkLocCache*>(CkLocalBranch(cacheID));
+  if (cache == nullptr)
```

Comment:
Could make this and the other error checking in this function conditionally compiled on CMK_ERROR_CHECKING

Reply:
Do you think that is worth doing for stuff like this that is only at creation of a new array? It's not a performance critical operation, and it's not like it's going to save overhead over a ton of very frequent calls.

Reply:
Yeah, it's fine in that kind of non-performance-critical case